hnn_core/params.py

```diff
@@ -655,6 +656,43 @@ def compare_dictionaries(d1, d2):
     return d1
 
 
+def _convert_to_path(value):
+    if isinstance(value, str):
```

Comment:
Why is this necessary? It's not really a proper typecheck since it will just return the same `value` if on non string is passed

Reply:
The function convert_to_path accepts both string and Path types. But it will convert string types to a Path object so it can use Path object methods like .suffix on 687. If neither are passed it will fail on the that line.

Reply:
I added the type checking and additional test we discussed yesterday. Should be ready to merge when tests pass.